cranelift/codegen/meta/src/isa/x86/instructions.rs

```diff
@@ -562,6 +562,7 @@ pub(crate) fn define(
             "#,
             &formats.unary_global_value,
         )
+        .clobbers_all_regs(true)
```

Comment:
`clobber_all` is a bit more pessimistic than we need for `__tls_get_addr`, which seems to really be `has the architectural effects of an ABI-conforming call` (callee-save registers are not clobbered)

Since you and @bjorn3 were both going for a `clobbers_all*` change this seems fine, but can I bug you for a comment here and below that this is a bit broader than should be necessary?

Reply:
Callee-saved registers are clobbered for normal calls too currently. There is a fixme in spilling.rs for that.

Reply:
I added a comment, please let me know what you think of it!